libraries/node-core-library/src/Import.ts

```diff
@@ -83,4 +128,105 @@ export class Import {
+
+    if (resolvePath.startsWith('.')) {
+      // This looks like a conventional relative path
+      return nodeJsPath.resolve(normalizedRootPath, resolvePath);
+    }
+
+    normalizedRootPath =
+      PackageJsonLookup.instance.tryGetPackageFolderFor(normalizedRootPath) || normalizedRootPath;
+
+    let slashAfterPackageNameIndex: number;
+    if (resolvePath.startsWith('@')) {
+      // This looks like a scoped package name
+      slashAfterPackageNameIndex = resolvePath.indexOf('/', resolvePath.indexOf('/') + 1);
+    } else {
+      slashAfterPackageNameIndex = resolvePath.indexOf('/');
+    }
+
+    let packageName: string;
+    let pathInsidePackage: string | undefined;
+    if (slashAfterPackageNameIndex === -1) {
+      // This looks like a package name without a path
+      packageName = resolvePath;
+    } else {
+      packageName = resolvePath.substr(0, slashAfterPackageNameIndex);
+      pathInsidePackage = resolvePath.substr(slashAfterPackageNameIndex + 1);
+    }
+
+    if (options.includeSystemModules === true) {
+      if (!Import._builtInModules) {
+        Import._builtInModules = new Set<string>(nodeModule.builtinModules);
+      }
+
+      // First, check resolvePath because some built-in modules have more than one slash in their name
+      if (Import._builtInModules.has(resolvePath)) {
+        return resolvePath;
+      } else if (Import._builtInModules.has(packageName)) {
+        if (pathInsidePackage !== undefined) {
+          throw new Error(
+            `The package name "${packageName}" resolved to a NodeJS system module, but the ` +
+              `path to resolve ("${resolvePath}") contains a path inside the system module, which is not allowed.`
+          );
+        }
+
+        return packageName;
+      }
+    }
+
+    let resolvedPackagePath: string | undefined;
+    if (options.allowSelfReference === true) {
+      // See if we're trying to resolve to the current package
+      const ownPackageJsonPath: string | undefined = PackageJsonLookup.instance.tryGetPackageJsonFilePathFor(
+        normalizedRootPath
+      );
+      if (
+        ownPackageJsonPath &&
+        PackageJsonLookup.instance.loadPackageJson(ownPackageJsonPath).name === packageName
+      ) {
+        resolvedPackagePath = nodeJsPath.dirname(ownPackageJsonPath);
+      }
+    }
+
+    if (!resolvedPackagePath) {
+      try {
+        resolvedPackagePath = nodeJsPath.dirname(
+          Resolve.sync(
+            // Append a slash to the package name to ensure `resolve.sync` doesn't attempt to return a system package
+            `${packageName}/`,
+            {
+              basedir: normalizedRootPath,
+              preserveSymlinks: false,
+              packageFilter: (pkg: { main: string }): { main: string } => {
+                // In case the "main" property isn't defined, set it to something we know will exist
+                pkg.main = 'package.json';
```

Comment:
The above comment is incorrect. There are two different kinds of resolving:

- `f1("jquery") --> "./node_modules/jquery/lib/index.js"`
- `f2("jquery") --> "./node_modules/jquery"`

The trick for `f2()` is to hardwire `pkg.main` to return `package.json` and then call `path.dirname()` on the result.  For `f1()` we can use `resolve.sync()` in the normal way.

Reply:
BTW a missing `"main"` field is normally NOT a problem, because if unspecified then Node.js [will default to](https://nodejs.org/api/modules.html#modules_all_together) looking for `./index.js`, `./index.json`, `index.node`.

@BPapp-MS  FYI